PythonVisualizations/Array.py

```diff
@@ -91,22 +98,32 @@ def insert(self, val):
             val, self.canvas.itemconfigure(cellPair[0], 'fill'), *cellPair))
         callEnviron ^= set(cellPair) # Remove new cell from temp call environ
 
+
         # advance index for next insert
         self.highlightCodeTags('increment_count', callEnviron)
-        self.moveItemsBy(indexDisplay, (self.CELL_SIZE, 0))
+        
+        # Move nItems pointer
+        self.moveItemsBy(self.nItems, (self.CELL_SIZE, 0))
         self.wait(0.1)
 
         self.highlightCodeTags([], callEnviron)
         self.cleanUp(callEnviron)
 
     def removeFromEnd(self):
+        callEnviron = self.createCallEnvironment()   
+               
         if len(self.list) == 0:
             self.setMessage('Array is empty!')
             return
-        callEnviron = self.createCallEnvironment()
+        

```

Comment:
Moving the call to `self.createCallEnvironment()` before the test for an empty list could cause a problem.  If the list is empty, `removeFromEnd()` returns without calling `cleanUp(callEnvrion)`.  That will leave an empty set on the call stack.  That could potentially cause a problem later when `cleanUp` decides how to clean up the bottomost call on the stack.

Instead, I think the call to `self.createCallEnvironment()` should be done after the test for the empty list and possible return.